csrc/ir_internal_nodes.h

```diff
@@ -1035,28 +1035,38 @@ class TORCH_CUDA_CU_API MmaOp : public Expr {
   //  after additional cleaning ups.
   struct OptionsInMma {
     MmaOptions::MacroType macro = MmaOptions::MacroType::NoMMA;
-    MmaOptions::MmaInputLayout operand_layout = MmaOptions::MmaInputLayout::TT;
     int accumulator_stride = 0;
 
     bool operator==(const OptionsInMma& other) const {
-      return macro == other.macro && operand_layout == other.operand_layout &&
+      return macro == other.macro &&
           accumulator_stride == other.accumulator_stride;
     }
   };
 
   using AxesData = std::vector<int>;
-  using MmaInputLayoutOpt = c10::optional<MmaOptions::MmaInputLayout>;
+  using MmaInputLayoutOpt = std::optional<MmaOptions::MmaInputLayout>;
   using Expr::Expr;
 
-  MmaOp(IrBuilderPasskey, Val* out, Val* in_a, Val* in_b, Val* init);
+  MmaOp(
+      IrBuilderPasskey,
+      Val* out,
+      Val* in_a,
+      Val* in_b,
+      Val* init,
+      const bool eval_attributes = true);
```

Comment:
What does this flag mean?

Reply:
The idea was to keep reusing basic constructor, but at the same time avoid possibly heavy attributes initialization if right after the analysis we replaced data with attributes passed to constructor (one with larger number of parameters).

Not the most graceful way of handling this so I redone it:
- `initializeCore`, will do most of the work that was initially in basic constructor:
  - inputs validation,
  - `Val*`s assignment,
  - allocated attributes,
- the basic constructor (only with `Val`s):
  - initialize core,
  - perform analysis, based on the inputs' type (if `Vals` are `TensorView`),
- the constructor with attributes:
  - initialize core,
  - initialize attributes with values passed as arguments,

So hack with default param is not needed anymore.